tests/unit/stream_alert_cli/test_config.py

```diff
@@ -14,67 +14,73 @@
 limitations under the License.
 """
 # pylint: disable=protected-access
-from contextlib import nested
 import json
 
 from mock import Mock, patch
 from nose.tools import assert_equal, assert_true
+from pyfakefs import fake_filesystem_unittest
 
 from stream_alert_cli.config import CLIConfig
-from tests.unit.helpers.base import basic_streamalert_config, mock_open
+from tests.unit.helpers.base import basic_streamalert_config
 
 
 class TestCLIConfig(object):
     """Test class for CLIConfig"""
+    # pylint: disable=no-self-use
 
     def __init__(self):
-        self.mocked_opens = None
+        self.fs_patcher = None
 
     def setup(self):
         """Setup before each method"""
         config_data = basic_streamalert_config()
-        self.mocked_opens = [
-            mock_open('conf/global.json', json.dumps(config_data['global'])),
-            mock_open('conf/lambda.json', json.dumps(config_data['lambda'])),
-            mock_open('conf/clusters/prod.json', json.dumps(config_data['clusters']['prod']))
-        ]
+
+        self.fs_patcher = fake_filesystem_unittest.Patcher()
+        self.fs_patcher.setUp()
+
+        self.fs_patcher.fs.CreateFile('/conf/global.json',
+                                      contents=json.dumps(config_data['global']))
+        self.fs_patcher.fs.CreateFile('/conf/lambda.json',
+                                      contents=json.dumps(config_data['lambda']))
+        self.fs_patcher.fs.CreateFile('/conf/clusters/prod.json',
+                                      contents=json.dumps(config_data['clusters']['prod']))
+
+    def teardown(self):
+        """Teardown after each method"""
+        self.fs_patcher.tearDown()
 
     def test_load_config(self):
         """CLI - Load config"""
-        with nested(*self.mocked_opens):
-            config = CLIConfig()
-            assert_equal(config['global']['account']['prefix'], 'unit-testing')
+        config = CLIConfig()
+        assert_equal(config['global']['account']['prefix'], 'unit-testing')
 
     @patch('logging.Logger.error')
     @patch('stream_alert_cli.config.CLIConfig.write')
     def test_toggle_metric(self, write_mock, log_mock):
         """CLI - Metric toggling"""
-        with nested(*self.mocked_opens):
-            config = CLIConfig()
+        config = CLIConfig()
```

Comment:
Since this is used in most (all?) test methods, can you add this as a instance property in the `__inti__` and `setup` to be used throughout the class tests?

Reply:
Hm, good point, but in this case CLIConfig() is the unit under test. Would it make more sense to leave it in each test case?

Reply:
That's fine - the `setup` method gets newly called with each method that runs anyway (effectively doing the same thing you're doing here).